### DISCUSSION TITLE
Azure DevOps CI Restore Failure - The element <#text> beneath element <ItemGroup> is unrecognized.

### DISCUSSION THREAD

**Discussion by jucchytil** [oqtane/oqtane.framework#961] [General]
I've copied the oqtane.framework-dev contents to an azure DevOps git repo and created a .Net Core CI Pipeline to build it. I added a Use .Net Core sdk 5.x step before it. 
![image](https://user-images.githubusercontent.com/17008787/100524186-a8f06f80-317b-11eb-90a4-1088ce559a5f.png)

It fails in the restore step.

>D:\a\1\s\Oqtane.Server\wwwroot\Modules\Templates\External\Client\[Owner].[Module].Client.csproj(27,3): error MSB4067: The element <#text> beneath element <ItemGroup> is unrecognized.

- Shouldn't this build OK?
- Is there a document somewhere that describes how to set up a build?

**Comment by jucchytil** [comment]
I was able to get past this error by having the build target the server project. Unfortunately, deploying the output from the server project did not produce a working web-app.  More research needed here.

**Comment by sbwalker** [comment]
The files in Modules\Templates should not be included as part of the build - they contain replaceable tokens and are only intended for scaffolding modules. If you look in the Server project file you will see these files are excluded.
